### DISCUSSION TITLE
Support graphics in `rio` terminal

### DISCUSSION THREAD

**Discussion by srid** [sxyazi/yazi#266] [4. Ideas]
Requires https://github.com/raphamorim/rio/issues/38

**Comment by sxyazi** [comment]
Hi, I'm interested in providing graphical support for `rio` terminal, but it seems that currently Yazi can only wait for `rio` to implement any one of the protocols like Kitty, iTerm2, or Sixel.

Alternatively, if you're using Linux, you can directly combine Yazi with Überzug++ to cover both X11 and Wayland desktop environments.

**Comment by raphamorim** [comment]
Hey @srid @sxyazi thanks for bringing it up. Would kitty image protocol be enough?

**Comment by sxyazi** [comment]
Yep, Kitty is enough.

I noticed that Rio also supports the Windows system. Due to the limitations of ConPTY, on Windows, it can only use the iTerm2 protocol. It would be even better if it could be also implemented.

I have tested iTerm2 protocol on WezTerm and Mintty, and it works properly on Windows. Of course, this is not absolutely necessary.

**Comment by raphamorim** [comment]
@sxyazi another question based on your comments,

does the sixel works on yazi for windows?

**Comment by sxyazi** [comment]
@raphamorim Nope, I can't get Sixel to work on Yazi for Windows. I've tested Mintty and WezTerm simultaneously, and none of them work.

BTW, in the case of Mintty, it's a bit special. If your program is built using Mingw GCC, just like the built-in `/bin/cat` in Mingw, then you can make it work *only in* Mintty.

See https://github.com/mintty/mintty/issues/1192 for more details.

**Comment by raphamorim** [comment]
Thank you @sxyazi for the explanation, so just to confirm sixel or kitty for unix and iterm2 for windows right?

**Comment by sxyazi** [comment]
Yeah that's it. You're welcome @raphamorim

**Comment by linsui** [comment]
https://raphamorim.io/rio/docs/features/iterm-image-protocol/ Rio supports iTerm protocol now and I can use imgcat in it. But I can't make yazi work in it by pretending iTerm or WezTerm.

**Reply by sxyazi** [reply]
That's so weird, I can't get it to work with my `imgcat`, I'm using the one provided by iTerm2, https://iterm2.com/utilities/imgcat.

I did a quick check, seems it's still in progress, and part of the Rio 0.1 plan, https://github.com/raphamorim/rio/issues/335

**Reply by linsui** [reply]
I tested it with https://github.com/trashhalo/imgcat (img-cat in nixpkgs). Oh, it seems this program doesn't use iTerm protocol. Sorry for the noise...

**Reply by sxyazi** [reply]
Still can't get it to work for IIP on Rio, I only see a pixelated mosaic image, which is a character-based image similar to [Chafa](https://github.com/hpjansson/chafa), rather than an IIP image.

**Reply by linsui** [reply]
Yes, it seems Rio hasn't supported iTerm protocol yet thought it says

> Images in Terminal
> The Rio can display images within the terminal using iTerm2 and kitty image protocols.

on the homepage. I can't get the kitty protocol working, either.

**Comment by linsui** [comment]
In the latest version rio has added sixel support.

**Comment by raphamorim** [comment]
Fyi: @sxyazi I plan to add iterm2 image protocol to next version. Please let me know any issue with sixel 🙏 

<img width="645" alt="Screenshot 2024-09-07 at 18 03 53" src="https://github.com/user-attachments/assets/367f9828-df80-41e6-88aa-83b3209fdc5f">

**Comment by sxyazi** [comment]
Image preview in the Rio terminal is now supported in https://github.com/sxyazi/yazi/pull/1690 🎉 

Thanks @raphamorim for the PR!
